cmd/agent/main.go

```diff
@@ -145,56 +145,55 @@ func runController(ctx context.Context, opts managerOpts) error {
 	}
 
 	// shared features between Ingress and Gateway (tunnels)
-	if opts.enableFeatureIngress || opts.enableFeatureGateway {
```

Comment:
For the agent pods, I removed this check because currently users would not be able to use `AgentEndpoints` if they opt-out of Ingress and Gateway API support and that seems wrong IMO. Using the `AgentEndpoint` resources (or any others) directly instead of through our translation layer should be a perfectly valid use-case.